src/openloco/windows/town_list.cpp

```diff
@@ -1,16 +1,1581 @@
+            {
+                registers regs;
+                regs.edx = widgetIndex;
+                regs.esi = (int32_t)self;
+                regs.edi = (int32_t)&self->widgets[widgetIndex];
+                call(0x0049AB72, regs);
+            }
+        }
+
+        // 0x0049B2B5
+        static void updateActiveThumb(window* self)
+        {
+            uint16_t scrollHeight = 0;
+            self->call_get_scroll_size(0, 0, &scrollHeight);
+            self->scroll_areas[0].v_bottom = scrollHeight;
+
+            auto i = 0;
+            for (; i <= self->var_83C; i++)
+            {
+                if (self->row_info[i] == self->row_hover)
+                    break;
+            }
+
+            if (i >= self->var_83C)
+                i = 0;
+
+            i = (i / 5) * rowHeight;
+
+            self->scroll_areas[0].v_top = i;
+
+            ui::scrollview::update_thumbs(self, widx::scrollview);
+        }
+
+        // 0x0049AF98
+        static void on_resize(window* self)
+        {
+            self->invalidate();
+            gfx::ui_size_t minWindowSize = { self->min_width, self->min_height };
+            gfx::ui_size_t maxWindowSize = { self->max_width, self->max_height };
+            bool hasResized = self->set_size(minWindowSize, maxWindowSize);
+            /*bool hasResized = false;
+            if ((self->height < minWindowSize.height) || (self->height > maxWindowSize.height))
+                hasResized = true;
+            self->set_size(minWindowSize, maxWindowSize);*/
+            if (hasResized)
+                updateActiveThumb(self);
+        }
+
+        // 0x0049AE83
+        static void get_scroll_size(ui::window* self, uint32_t scrollIndex, uint16_t* scrollWidth, uint16_t* scrollHeight)
+        {
+            *scrollHeight = (4 + self->var_83C) / 5;
+            if (*scrollHeight == 0)
+                *scrollHeight += 1;
+            *scrollHeight *= rowHeight;
+        }
+
+        // 0x0049ABBB
+        static void tooltip(FormatArguments& args, ui::window* window, widget_index widgetIndex)
+        {
+            args.push(string_ids::tooltip_scroll_building_list);
+        }
+
+        // 0x0042DB95
+        static void drawBuildingThumb(gfx::drawpixelinfo_t* clipped, const openloco::building_object* buildingObj, uint8_t buildingRotation, int16_t x, int16_t y, colour_t colour)
+        {
+            registers regs;
+            regs.cx = x;
+            regs.dx = y;
+            regs.esi = colour;
+            regs.eax = buildingRotation;
+            regs.edi = (uint32_t)clipped;
+            regs.ebp = (uint32_t)buildingObj;
+            call(0x0042DB95, regs);
+        }
+
+        // 0x0049AA1C
+        static void draw_scroll(ui::window* self, gfx::drawpixelinfo_t* dpi, uint32_t scrollIndex)
+        {
+            auto shade = colour::get_shade(self->colours[1], 3);
+            gfx::clear_single(*dpi, shade);
+
+            loco_global<uint16_t, 0x01135C58> word_1135C58;
```

Comment:
this variable is only used in this function just make it a local variable. No need to use a loco global.